<a href="https://colab.research.google.com/github/LucasMartinscode/Python/blob/main/Winedataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Objetivo**

---

Descobrir inferências no preço de vinhos em dolares e por pontos de cada vinho utilizando machine learn, usando o conceito de tendência central, dispersão, mínimo e máximo. 

**Métodos usados**

*   Funções no Python
*   Implementação de bancos de dados
*   Estátistica descritiva
*   Queries de python
*   Visualização de dados

**Stack**

*   Google Colab
*   Pandas
*   Random
*   Matplotlib
*   Plotly


**Data Sources**

*   Banco de dados encontrado no Kaggle


In [ ]:
import pandas as pd #Biblioteca para análise e manipulação de dados
import random 
import plotly.graph_objects as go #Bibliotecas para análise com gráficos.
import plotly.express as px

dataset = pd.read_csv('winemag-data-130k-v2.csv') #Lendo o dataset.
datasetdropna = dataset.dropna() #Eliminando valores nulos junto as linhas tabela.

In [ ]:
datasetdropna.head() #Imprimindo os 10 datasets não nulos.

Unnamed: 0 country                                        description  \
4            4      US  Much like the regular bottling from 2012, this...   
10          10      US  Soft, supple plum envelopes an oaky structure ...   
23          23      US  This wine from the Geneseo district offers aro...   
25          25      US  Oak and earth intermingle around robust aromas...   
35          35      US  As with many of the Erath 2010 vineyard design...   

                           designation  points  price    province  \
4   Vintner's Reserve Wild Child Block      87   65.0      Oregon   
10                      Mountain Cuvée      87   19.0  California   
23                 Signature Selection      87   22.0  California   
25                 King Ridge Vineyard      87   69.0  California   
35                              Hyland      86   50.0      Oregon   

             region_1           region_2     taster_name  \
4   Willamette Valley  Willamette Valley    Paul Gregutt   
10        Napa Valley               Napa  Virginie Boone   
23        Paso Robles      Central Coast   Matt Kettmann   
25       Sonoma Coast             Sonoma  Virginie Boone   
35        McMinnville  Willamette Valley    Paul Gregutt   

   taster_twitter_handle                                              title  \
4            @paulgwine   Sweet Cheeks 2012 Vintner's Reserve Wild Child...   
10               @vboone  Kirkland Signature 2011 Mountain Cuvée Caberne...   
23         @mattkettmann  Bianchi 2011 Signature Selection Merlot (Paso ...   
25               @vboone  Castello di Amorosa 2011 King Ridge Vineyard P...   
35           @paulgwine          Erath 2010 Hyland Pinot Noir (McMinnville)   

               variety               winery  
4           Pinot Noir         Sweet Cheeks  
10  Cabernet Sauvignon   Kirkland Signature  
23              Merlot              Bianchi  
25          Pinot Noir  Castello di Amorosa  
35          Pinot Noir                Erath

Amostragem Aleatória simples

In [ ]:
Amostra_aleatoria_simples = datasetdropna.sample(n= 100, random_state=2) # Amostra simples utilizando 100 linhas usando random state para ser variado o número de amostras

Amostra_aleatoria_simples.head()

Unnamed: 0 country                                        description  \
87382        87382      US  Light and fruity, this smells like cucumbers, ...   
26520        26520      US  This is the producer's inaugural offering of t...   
57871        57871      US  This mashup of 62% Syrah, 16% Petite Sirah, 8%...   
108738      108738      US  Very rich aromatically in golden Santa Rosa pl...   
21736        21736      US  Gréine is a blend of Syrah and Petite Sirah, w...   

                      designation  points  price    province  \
87382                         Gia      86   17.0  California   
26520             Pellet Vineyard      93   75.0  California   
57871                   Reckoning      93   45.0  California   
108738  Indindoli Family Vineyard      89   33.0  California   
21736                      Gréine      84   38.0  Washington   

                    region_1          region_2     taster_name  \
87382             California  California Other      Jim Gordon   
26520            Napa Valley              Napa  Virginie Boone   
57871            Paso Robles     Central Coast   Matt Kettmann   
108738  Russian River Valley            Sonoma  Virginie Boone   
21736   Columbia Valley (WA)   Columbia Valley    Paul Gregutt   

       taster_twitter_handle  \
87382       @gordone_cellars   
26520                @vboone   
57871          @mattkettmann   
108738               @vboone   
21736            @paulgwine    

                                                    title             variety  \
87382   Francis Ford Coppola 2013 Gia Pinot Grigio (Ca...        Pinot Grigio   
26520   Pellet Estate 2014 Pellet Vineyard Merlot (Nap...              Merlot   
57871                ONX 2012 Reckoning Red (Paso Robles)           Red Blend   
108738  Most Wanted Winery 2009 Indindoli Family Viney...          Chardonnay   
21736   Cairdeas 2011 Gréine Syrah-Petite Sirah (Colum...  Syrah-Petite Sirah   

                      winery  
87382   Francis Ford Coppola  
26520          Pellet Estate  
57871                    ONX  
108738    Most Wanted Winery  
21736               Cairdeas

**Criando tabelas usando a biblioteca plotly e cálculos de estátistica descritiva por provincia.**



*   Cálculos estatísticos como média, desvio padrão, mín e max. 




In [ ]:
media_price = Amostra_aleatoria_simples['price'].mean() #Descobrindo a média total de preços
desvio_padrao_price = Amostra_aleatoria_simples['price'].std() #Descobrindo o desvio padrão total de preços
min_price = Amostra_aleatoria_simples['price'].min() #Descobrindo o minimo total de preços
max_price = Amostra_aleatoria_simples['price'].max() #Descobrindo o maximo total de preços

media_points = Amostra_aleatoria_simples['points'].mean() #Descobrindo a média total de pontos
desvio_padrao_points = Amostra_aleatoria_simples['points'].std() #Descobrindo o desvio padrão total de pontos
min_points = Amostra_aleatoria_simples['points'].min() #Descobrindo o minimo total de pontos
max_points = Amostra_aleatoria_simples['points'].max() #Descobrindo o maximo total de pontos

Medidas_totais_table = {                                      #Criando uma tabela para medidas totais de todas regiões retiradas na amostra.
    'Medida': ['Média', 'Desvio Padrão', 'Mínimo', 'Máximo'],
    'Medidas totais de preços': [media_price, desvio_padrao_price, min_price, max_price],
    'Medidas totais de pontos': [media_points, desvio_padrao_points, min_points, max_points]}

Tabela_resultados = pd.DataFrame(Medidas_totais_table)

Tabela_resultados.head()

Medida  Medidas totais de preços  Medidas totais de pontos
0          Média                 42.880000                 89.650000
1  Desvio Padrão                 28.740314                  2.875796
2         Mínimo                  9.000000                 80.000000
3         Máximo                185.000000                 97.000000



*   Criando medidas de cada provincia




In [ ]:
dados_california = Amostra_aleatoria_simples[Amostra_aleatoria_simples['province'] == 'California'] #Filtrando medidas apenas da California 

media_price_california = dados_california['price'].mean()
desvio_padrao_price_california = dados_california['price'].std()
min_price_california = dados_california['price'].min()
max_price_california = dados_california['price'].max()

media_points_california = dados_california['points'].mean()
desvio_padrao_points_california = dados_california['points'].std()
min_points_california = dados_california['points'].min()
max_points_california = dados_california['points'].max()


dados_oregon = Amostra_aleatoria_simples[Amostra_aleatoria_simples['province'] == 'Oregon'] #Filtrando medidas apenas de Oregon

media_price_oregon = dados_oregon['price'].mean()
desvio_padrao_price_oregon = dados_oregon['price'].std()
min_price_oregon = dados_oregon['price'].min()
max_price_oregon = dados_oregon['price'].max()

media_points_oregon = dados_oregon['points'].mean()
desvio_padrao_points_oregon = dados_oregon['points'].std()
min_points_oregon = dados_oregon['points'].min()
max_points_oregon = dados_oregon['points'].max()

dados_washington = Amostra_aleatoria_simples[Amostra_aleatoria_simples['province'] == 'Washington'] #Filtrando medidas apenas de Washington

media_price_washington = dados_washington['price'].mean()
desvio_padrao_price_washington = dados_washington['price'].std()
min_price_washington = dados_washington['price'].min()
max_price_washington = dados_washington['price'].max()

media_points_washington = dados_washington['points'].mean()
desvio_padrao_points_washington = dados_washington['points'].std()
min_points_washington = dados_washington['points'].min()
max_points_washington = dados_washington['points'].max()

Medidas_isoladas = {                                         #Criando uma tabela isolada para medidas de cada região retiradas na amostra.
    'Medida': ['Média', 'Desvio Padrão', 'Mínimo', 'Máximo'],
    'Medidas de preços da California': [media_price_california, desvio_padrao_price_california, min_price_california, max_price_california],
    'Medidas de preços de Washington': [media_price_washington, desvio_padrao_price_washington, min_price_washington, max_points_washington],
    'Medidas dos preços de Oregon': [media_price_oregon, desvio_padrao_price_oregon, min_price_oregon, max_price_oregon],
    'Medidas de pontos da California': [media_points_california, desvio_padrao_points_california, min_points_california, max_points_california],
    'Medidas de pontos de Washington': [media_points_washington, desvio_padrao_points_washington, min_points_washington, max_points_washington],
    'Medidas dos pontos de Oregon': [media_points_oregon, desvio_padrao_points_oregon, min_points_oregon, max_points_oregon]}

Tabela_resultados = pd.DataFrame(Medidas_isoladas)

Tabela_resultados.head()

Medida  Medidas de preços da California  \
0          Média                        45.018868   
1  Desvio Padrão                        30.322935   
2         Mínimo                        13.000000   
3         Máximo                       175.000000   

   Medidas de preços de Washington  Medidas dos preços de Oregon  \
0                        36.933333                     46.705882   
1                        15.165599                     40.108859   
2                         9.000000                     14.000000   
3                        93.000000                    185.000000   

   Medidas de pontos da California  Medidas de pontos de Washington  \
0                        89.415094                        89.500000   
1                         3.248869                         2.177471   
2                        80.000000                        84.000000   
3                        97.000000                        93.000000   

   Medidas dos pontos de Oregon  
0                     90.647059  
1                      2.644361  
2                     85.000000  
3                     94.000000



*   Criando gráfico de barras comparando as médias de preços de vinhos entre as províncias da Califórnia, Washington e Oregon.




In [ ]:
provincias = ['California', 'Washington', 'Oregon']      #Criando dicionários de dados

medias = [media_price_california, media_price_washington, media_price_oregon]

cores_barras = ['blue', 'green', 'red']

fig1 = go.Figure(data=go.Bar(x=provincias, y=medias, marker=dict(color=cores_barras)))    #Criando um gráfico de barras com média de preços de vinhos por região

fig1.update_layout(
    title='Média dos Preços do Vinho por Província',
    xaxis=dict(title='Província'),
    yaxis=dict(title='Média dos preços do Vinho'), 
    plot_bgcolor='white'
)

fig1.show()

fig1.write_html('Gráfico de barras.html')


*   Criando um gráfico de dispersão para plotar os pontos do vinho em relação aos preços. Cada ponto representa um vinho.

In [125]:

fig2 = px.scatter(Amostra_aleatoria_simples, x='price', y='points', color='province') # Criando um gráfico estatístico com cores por província


fig2.update_layout(
    title='Dispersão de Preço vs. Pontos por Região',
    xaxis=dict(title='Preço'),
    yaxis=dict(title='Pontos'),
    plot_bgcolor='white'
)

fig2.show()

![Gráfico de dispersão](https://raw.githubusercontent.com/LucasMartinscode/Python/main/Gr%C3%A1fico%20de%20dipers%C3%A3o.html)](https://github.com/LucasMartinscode/Python/blob/main/Gr%C3%A1fico%20de%20dispers%C3%A3o.html)

/bin/bash: -c: line 0: syntax error near unexpected token `('
/bin/bash: -c: line 0: `[Gráfico de dispersão](https://raw.githubusercontent.com/LucasMartinscode/Python/main/Gr%C3%A1fico%20de%20dipers%C3%A3o.html)](https://github.com/LucasMartinscode/Python/blob/main/Gr%C3%A1fico%20de%20dispers%C3%A3o.html)'




*   Criando um gráfico BoxPlot para visualização de mín e máx de cada região.




In [ ]:
fig3 = px.box(Amostra_aleatoria_simples, x='province', y='price', color='province', title='Distribuição de Preços por Província', # Criando o gráfico de box plot com mínimo e máximo por região e cores diferentes
             points="all", range_y=[Amostra_aleatoria_simples['price'].min(), Amostra_aleatoria_simples['price'].max()])


fig3.update_layout(                          # Configurando layout do gráfico
    xaxis=dict(title='Província'),
    yaxis=dict(title='Preço'),
    showlegend=False,
    plot_bgcolor='white'
)

fig3.show()
fig3.write_html('Gráfico boxplot.html')

**Conclusão**

Com base nos valores da média de todas regiões (42.88), desvio padrão de todas regiões (30.322935), mín e máx (13.00) e (175.00) respectivamente, podemos ter algumas inferências sobre os preços dos vinhos no futuro:

A média (42.88) pode ser considerada um valor central ou estimativa do preço médio dos vinhos. Isso significa que, em média, os preços dos vinhos podem se situar em torno desse valor no futuro.

O desvio padrão (30.322935) indica a dispersão dos preços em relação à média. Um desvio padrão maior sugere uma maior variabilidade nos preços, o que significa que os preços dos vinhos podem ter flutuações mais significativas no futuro.

O valor mínimo (13.00) representa o menor preço observado nos dados históricos. Isso indica que, no passado, houve vinhos com preços tão baixos quanto 13.00.

O valor máximo (175.00) representa o maior preço observado nos dados históricos. Isso indica que, no passado, houve vinhos com preços tão altos quanto 175.00.